report/sections/Chapter 11 - Reflection/Project-management.tex

```diff
@@ -0,0 +1,45 @@
+In the following section we will describe how the project was managed, which techniques were used, as well as which issues we encountered. Throughout we will reflect on the decisions that were made in this context.
+
+\section{Task management}
+One of the primary objectives in any project is to manage the tasks that are generated throughout the project. The goal is to have the ability to maintain an overview of what has been done, what needs to be done and how the tasks are distributed between developers. Additionally, project managers also need often want to be able to estimate how long tasks may take to complete and their difficulty. To accomplish this, projects may utilize several project management tools and techniques. In this section we will focus on the tools that we utilized, while in section \ref{sec:agile-dev} we will focus on the techniques that were utilized to manage the project.
+
+\subsection{Github projects}
+Initially we decided on using a tool called Github Projects. Github Projects is a project management tool that integrates issues, which is a text description of bugs and features that should be included in the project as well as pull requests into a board similar to a Kanban Board. This meant that we could create a board that was categorized into four different types: \textit{Todo}, \textit{In progress}, \textit{Ready for review} and \textit{Ready}, giving us a clear overview of what needed to be done, what was being worked on, what needed to be reviewed and what was done. All accessed from the same website. 
+
+While this tool offers a lot of benefits for developers in terms of project management, the tool was unfamiliar to all of the group members, as no member had any experience with the tool. The lack of experience became a hurdle within a few weeks of the project starting. This was due to several factors, including an unwillingness to understand the tool, but the primary issue that we were also learning other tools, frameworks and even a new programming language. All of which were more important to the project and therefore learning Github Projects became a less of a priority. To the point that we completely stopped using it, despite its promises.
```

Comment:
Probably not great to write that we were unwilling. Rather, something like `the learning curve added friction to adopting the tool, which added extraneous overhead to existing developer habits. This prevented group members from properly using the tool, and therefore made it a net negative utility.`

Reply:
This also supports the below sentence, about us reverting to tools we have experience with.

Reply:
Would also like to see a deeper explanation of what part of GitHub Projects added the extra overhead.
The descriptions made the tools sound almost identical, so perhaps we can clarify a bit?

Here are my own thoughts on the matter:
- GH projects would be easier if we were more accustomed to working with issues in our git commits, as much of the automation comes from referencing issues in commits.
- However, when you create a PR without doing all of this, you have to manually assign things. And a PR could be a subtask of a 'story task', which is what our task board has. That made it a bit more annoying.
- Managing a project generally has more 'fluffy' tasks than what can be represented only in the development domain. We often have tasks unrelated to code, which is hard to track with GH projects.

Reply:
```
GitHub Projects seemed promising, as it integrates with existing developer workflows. You track issues, and as you work on them, they flow from the backlog to in progress, to in review, and then is at last categorized as completed. When you submit a pull request, it goes to the review section, so other members are aware that it is pending. And when that pull request is merged, it goes to completed automatically.
Intuitively, it makes sense to track issues like this, as it is so closely aligned with the modern Git workflow on GitHub. 
However, we found that some members were not accustomed to this way of working, and as such, didn't take the extra steps to facilitate the automation, e.g. marking issues as related to the project.
And when you create a pull request without referencing issues related to the project, you have to manually assign things later. Given that a pull request doesn't always correlate directly with issues, this is easily forgotten.
Managing a project generally includes managing non-technical tasks as well, which is hard to track with Projects, as it is very development-specific.
The final straw was bugs where the automation workflows weren't invoked, resulting in confusion. This resulted in us not using the tool.
```